osu.Game.Rulesets.Taiko/Mods/TaikoModAlternate.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.Taiko.Objects;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public class TaikoModAlternate : ModAlternate<TaikoHitObject, TaikoAction>
+    {
+        [SettingSource("Playstyle", "Preferred Alternate Playstyle")]
+        public Bindable<Playstyle> Style { get; } = new Bindable<Playstyle>();
+
+        private TaikoAction? lastAction1;
+        private TaikoAction? lastAction2;
+        private TaikoAction lastAction;
+        private TaikoHitObject lastHitObject;
+        private double lastActionTime = 0;
+        private const double strong_hit_window = 30;
+
+        protected override void ResetActionStates()
+        {
+            lastAction1 = lastAction2 = null;
+        }
+
+        protected override bool OnPressed(TaikoAction action)
+        {
+            var blockInput = false;
+
+            var hitObject = HitObjects.FirstOrDefault((h) =>
+            {
+                var time = Interceptor.Time.Current;
+                var window = h.HitWindows.WindowFor(HitResult.Miss);
+                var endTime = (h as IHasDuration)?.EndTime ?? h.StartTime;
+                return (time > h.StartTime - window) && (time < endTime + window);
+            });
+
+            if (hitObject?.IsStrong ?? false)
```

Comment:
I'd extract the `if` branch to a method (`handleStrongHit`) to reduce nesting & complexity.